python/whylogs/experimental/core/udf_schema.py

```diff
@@ -0,0 +1,208 @@
+    """
+    Decorator to easily configure UDFs for your data set. Decorate your UDF
+    functions, then call generate_udf_dataset_schema() to create a UdfSchema
+    that includes the UDFs configured by your decorator parameters. The decorated
+    function will automatically be a UDF in the UdfSchema.
+
+    Specify udf_name to give the output of the UDF a name. udf_name
+    defautls to the name of the decorated function. Note that all lambdas are
+    named "lambda", so omitting udf_name on more than one lambda will result
+    in name collisions.
+
+    If any metrics are passed via the metrics argument, they will be attached
+    to the column produced by the UDF via the schema returned by generate_udf_dataset_schema().
+    If metrics is None, the UDF output column will get the metrics determined by
+    the other resolvers passed to generate_udf_dataset_schema(), or the STANDARD_RESOLVER
+    by default.
+    """
+
+    def decorator_register(func):
+        global _multicolumn_udfs, _resolver_specs
+        name = udf_name or func.__name__
+        _multicolumn_udfs.append(UdfSpec(col_names, {name: func}))
+        if metrics:
+            _resolver_specs.append(ResolverSpec(name, None, deepcopy(metrics)))
+
+        return func
+
+    return decorator_register
+
+
+def generate_udf_specs(other_udf_specs: Optional[List[UdfSpec]] = None) -> List[UdfSpec]:
+    """
+    Generates a list UdfSpecs that implement the UDFs specified
+    by the @register_dataset_udf decorators. You can provide a list of
+    other_udf_specs to include in addition to those UDFs registered via
+    the decorator.
+
+    For example:
+
+    @register_dataset_udf(col_names=["col1"])
+    def add5(x):
+        return x + 5
+
+    schema = UdfSchema(STANDARD_RESOLVER, udf_specs=generate_udf_specs())
+    why.log(data, schema=schema)
+
+    This will attach a UDF to column "col1" that will generate a new column
+    named "add5" containing the values in "col1" incremented by 5. Since these
+    are appended to the STANDARD_RESOLVER, the default metrics are also tracked
+    for every column.
+    """
+    specs = list(other_udf_specs) if other_udf_specs else []
+    specs += _multicolumn_udfs
+    return specs
+
+
+def generate_udf_dataset_schema(
+    other_udf_specs: Optional[List[UdfSpec]] = None,
+    resolvers: Optional[List[ResolverSpec]] = None,
+    types: Optional[Dict[str, Any]] = None,
+    default_config: Optional[MetricConfig] = None,
+    type_mapper: Optional[TypeMapper] = None,
+    cache_size: int = 1024,
+    schema_based_automerge: bool = False,
+    segments: Optional[Dict[str, SegmentationPartition]] = None,
+    validators: Optional[Dict[str, List[Validator]]] = None,
+) -> UdfSchema:
+    """
+    Returns a UdfSchema that implements any registered UDFs, along with any
+    other_udf_specs or resovlers passed in.
+    """
+
+    resolver_specs = resolvers + _resolver_specs if resolvers is not None else STANDARD_RESOLVER + _resolver_specs
```

Comment:
We had people not want frequent items tracked by default on the original features. The standard resolver is not a good general default for udf scenarios. We need a new default config that is standard -= frequent items.

This might be ok to have as an option, but I want the privacy preserving version as a helper method too.